gtsfm/scene_optimizer.py

```diff
@@ -233,46 +147,63 @@ def create_computation_graph(
+            num_images=num_images,
+            keypoints_list=keypoints_list,
+            i2Ri1_dict=i2Ri1_dict,
+            i2Ui1_dict=i2Ui1_dict,
+            v_corr_idxs_dict=v_corr_idxs_dict,
+            all_intrinsics=camera_intrinsics,
+            absolute_pose_priors=absolute_pose_priors,
+            relative_pose_priors=relative_pose_priors,
+            two_view_reports_dict=two_view_reports,
+            cameras_gt=cameras_gt,
+            gt_wTi_list=gt_wTi_list,
+            output_root=self.output_root,
         )
         if view_graph_two_view_reports is not None:
-            two_view_reports_dict[VIEWGRAPH_REPORT_TAG] = view_graph_two_view_reports
+            two_view_reports_post_viewgraph_estimator = view_graph_two_view_reports
 
         # Persist all front-end metrics and their summaries.
         # TODO(akshay-krishnan): this delays saving the frontend reports until MVO has completed, not ideal.
         metrics_graph_list: List[Delayed] = []
         annotation = dask.annotate(workers=self._output_worker) if self._output_worker else dask.annotate()
         with annotation:
-            for tag, report_dict in two_view_reports_dict.items():
-                delayed_results.append(
-                    dask.delayed(save_full_frontend_metrics)(
-                        report_dict,
-                        image_graph,
-                        filename="two_view_report_{}.json".format(tag),
-                        matching_regime=self.retriever._matching_regime,
-                        metrics_path=self._metrics_path,
-                        plot_base_path=self._plot_base_path,
-                    )
+            delayed_results.append(
+                dask.delayed(save_full_frontend_metrics)(
+                    two_view_reports,
+                    images,
+                    filename="two_view_report_{}.json".format(POST_ISP_REPORT_TAG),
+                    matching_regime=self.retriever._matching_regime,
+                    metrics_path=self._metrics_path,
+                    plot_base_path=self._plot_base_path,
                 )
-                metrics_graph_list.append(
-                    dask.delayed(two_view_estimator.aggregate_frontend_metrics)(
-                        report_dict,
-                        self._pose_angular_error_thresh,
-                        metric_group_name="verifier_summary_{}".format(tag),
-                    )
+            )
+            metrics_graph_list.append(
+                dask.delayed(two_view_estimator.aggregate_frontend_metrics)(
+                    two_view_reports,
+                    self._pose_angular_error_thresh,
+                    metric_group_name="verifier_summary_{}".format(POST_ISP_REPORT_TAG),
+                )
+            )
+
+            delayed_results.append(
+                dask.delayed(save_full_frontend_metrics)(
+                    two_view_reports_post_viewgraph_estimator,
+                    images,
```

Comment:
it does not need the images, only the image filenames. how long do we keep the images in memory? I think this futures design should make it easy to map images to image shapes and image filenames, and only keep those in memory, discarding the matrix.

Reply:
Agreed, but I would like to do it in a different PR. This PR is getting unmanageable now.

Reply:
sounds good, could you add a TODO/issue?

Reply:
Done